link/src/main/java/com/stripe/android/link/ui/verification/VerificationScreen.kt

```diff
@@ -0,0 +1,137 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.material.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.stripe.android.link.LinkActivityContract
+import com.stripe.android.link.R
+import com.stripe.android.link.theme.DefaultLinkTheme
+import com.stripe.android.link.theme.linkTextFieldColors
+import com.stripe.android.ui.core.elements.SectionCard
+
+@Preview
+@Composable
+private fun VerificationBodyPreview() {
+    DefaultLinkTheme {
+        VerificationBody(
+            redactedPhoneNumber = "+1********23",
+            onCodeEntered = { },
+            onResendCodeClick = { }
+        )
+    }
+}
+
+@Composable
+internal fun VerificationBody(
+    application: Application,
+    starterArgsSupplier: () -> LinkActivityContract.Args
+) {
+    val viewModel: VerificationViewModel = viewModel(
+        factory = VerificationViewModel.Factory(
+            application,
+            starterArgsSupplier
+        )
+    )
+
+    VerificationBody(
+        redactedPhoneNumber = viewModel.linkAccount.redactedPhoneNumber,
+        onCodeEntered = viewModel::onVerificationCodeEntered,
+        onResendCodeClick = viewModel::onResendCodeClicked
+    )
+}
+
+@Composable
+internal fun VerificationBody(
+    redactedPhoneNumber: String,
+    onCodeEntered: (String) -> Unit,
+    onResendCodeClick: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 20.dp, vertical = 20.dp),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Text(
+            text = stringResource(R.string.verification_header),
+            modifier = Modifier
+                .padding(vertical = 4.dp),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.h2
+        )
+        Text(
+            text = stringResource(R.string.verification_message, redactedPhoneNumber),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 4.dp, bottom = 30.dp),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.body1
+        )
+        VerificationCodeInput(onCodeEntered)
```

Comment:
Cool, I was hoping this would be a component we could reuse.  This is a field required for some of the upcoming LPMs.